Author: Upstream
Description: Report and fix both by Nils Carlson.
 Replaced a cc==0 check with proper Tcl_Eof() check.
Last-Modified: Fri, 23 Oct 2015 11:09:07 +0300
Bug: http://sourceforge.net/p/expect/patches/18/
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=799301

--- a/expect.c
+++ b/expect.c
@@ -1860,19 +1860,11 @@
     if (cc == EXP_DATA_NEW) {
 	/* try to read it */
 	cc = expIRead(interp,esPtr,timeout,tcl_set_flags);
-	
-	/* the meaning of 0 from i_read means eof.  Muck with it a */
-	/* little, so that from now on it means "no new data arrived */
-	/* but it should be looked at again anyway". */
-	if (cc == 0) {
+
+	if (Tcl_Eof(esPtr->channel)) {
 	    cc = EXP_EOF;
-	} else if (cc > 0) {
-	    /* successfully read data */
-	} else {
-	    /* failed to read data - some sort of error was encountered such as
-	     * an interrupt with that forced an error return
-	     */
 	}
+
     } else if (cc == EXP_DATA_OLD) {
 	cc = 0;
     } else if (cc == EXP_RECONFIGURE) {
